includes/class-wc-session-handler.php

```diff
@@ -253,6 +256,10 @@ public function save_data() {
 
 			wp_cache_set( $this->get_cache_prefix() . $this->_customer_id, $this->_data, WC_SESSION_CACHE_GROUP, $this->_session_expiration - time() );
 			$this->_dirty = false;
+			$allow_guest_session_removal = apply_filters( 'woocommerce_persistent_cart_enabled', true ) ? true : (bool) get_user_meta( get_current_user_id(), '_woocommerce_load_saved_cart_after_login', true );
```

Comment:
Could you please elaborate on why the guest session is only deleted if `woocommerce_persistent_cart_enabled` is `true` or if the user meta `_woocommerce_load_saved_cart_after_login` is `true`? It is not clear to me why this check is needed (I'm not saying it is not, just looking for clarification).

Reply:
You are correct @rodrigoprimo. Thanks! I noticed that without those checks the cart didn't work correctly. However, it seems that by reviewing the code I was able to leave them out by changing the `$old_session_key` comparison to loose type. I made the final pull request and have tested out the this change with both the `woocommerce_persistent_cart_enabled` filter true & false and using both guest users and logged in users. Thanks for your help again.